inventory/any.cf

```diff
@@ -435,6 +443,118 @@ bundle agent cfe_autorun_inventory_cpuinfo
+
+      "ec2_instance" -> { "CFE-2924" }
+        expression => regcmp(".*[aA]mazon.*", "$(cfe_autorun_inventory_dmidecode.dmi[bios-version])"),
+        scope => "namespace",
+        if => isvariable("cfe_autorun_inventory_dmidecode.dmi[bios-version]");
+
+      "ec2_instance" -> { "CFE-2924" }
+        expression => regcmp(".*[aA]mazon.*", "$(cfe_autorun_inventory_dmidecode.dmi[bios-vendor])"),
+        scope => "namespace",
+        if => isvariable("cfe_autorun_inventory_dmidecode.dmi[bios-vendor]");
+
+      "ec2_instance" -> { "CFE-2924" }
+        expression => regline( "^ec2.*", "/sys/hypervisor/uuid" ),
+        scope => "namespace",
+        if => fileexists("/sys/hypervisor/uuid");
+
+  reports:
+    (DEBUG|DEBUG_inventory_aws)::
+      "DEBUG $(this.bundle)";
+
+      "$(const.t)+ec2_instance"
+        if => "ec2_instance";
+}
+
+bundle agent cfe_autorun_inventory_aws_ec2_metadata
+# @brief Inventory ec2 metadata
+# Provides:
+{
+  methods:
+    !(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+      "cfe_autorun_inventory_aws_ec2_metadata_cache";
+      "cfe_autorun_inventory_aws_ec2_metadata_from_cache";
+}
+bundle agent cfe_autorun_inventory_aws_ec2_metadata_cache
+# @brief Cache ec2 metadata from http request for one day
+#
+# Provides cache of ec2 instance metadata for inventory
+{
+
+  vars:
+
+    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+
+      "curl"      string => "$(paths.curl)";
+      "curl_opts" string => "--max-time 1 --connect-timeout 1 --silent";
+      "URL"       string => "http://169.254.169.254/latest/dynamic/instance-identity/document";
+      "cache"     string => "$(sys.statedir)/aws_ec2_metadata";
+
+  commands:
+
+    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+
+      "$(curl)"
+        args => "$(curl_opts) $(URL) -o $(cache)",
+        if => not( fileexists( $(cache) ));
+
+  files:
+    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+      "$(cache)"
+        delete => tidy,
+        file_select => days_old(1);
+}
+
+bundle agent cfe_autorun_inventory_aws_ec2_metadata_from_cache
+# @brief Inventory ec2 metadata from cache
+#
+# Provides inventory for EC2 Region, EC2 instanceId, EC2 instanceType, EC2
+# imageID, and EC2 availabilityZone
+{
+  classes:
+
+    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+
+      "have_cached_instance_identity"
+        expression => fileexists( $(cfe_autorun_inventory_aws_ec2_metadata_cache.cache) );
+
+  vars:
+
+    have_cached_instance_identity.ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
+
+      "data" data => readjson( $(cfe_autorun_inventory_aws_ec2_metadata_cache.cache), 100K);
+
+      "region" string => "$(data[region])", meta => { "inventory", "attribute_name=EC2 region" };
+      "instanceId" string => "$(data[instanceId])", meta => { "inventory", "attribute_name=EC2 instanceId" };
```

Comment:
I think the user facing attribute name should be `EC2 Instance ID`(no camelcase). This matches how it's represented in the AWS EC2 Management console. Same for others below.

Reply:
OK, thanks. Will do.

And yes, ack the commit history cleanup.

Reply:
DONE